archive/tar.go

```diff
@@ -20,55 +20,62 @@ func NewTarArchive() Archive {
 	return &tarArchive{}
 }
 
-func (a *tarArchive) Pack(src string, w io.Writer) error {
-	// ensure the src actually exists before trying to tar it
-	if _, err := os.Stat(src); err != nil {
-		return err
-	}
-
+func (a *tarArchive) Pack(srcs []string, w io.Writer) error {
 	tw := tar.NewWriter(w)
 	defer tw.Close()
 
-	// walk path
-	return filepath.Walk(src, func(path string, fi os.FileInfo, err error) error {
-		if err != nil {
+	// Loop through each source
+	var fwErr error
+	for _, s := range srcs {
+		// ensure the src actually exists before trying to tar it
+		if _, err := os.Stat(s); err != nil {
 			return err
 		}
 
-		var link string
-		if fi.Mode()&os.ModeSymlink == os.ModeSymlink {
-			if link, err = os.Readlink(path); err != nil {
+		// walk path
+		fwErr = filepath.Walk(s, func(path string, fi os.FileInfo, err error) error {
```

Comment:
Looks like `fwErr` will be overwritten on each iteration of the loop so it will only return the last value set.